### PR TITLE
[work in progress] [apm] Clarify how data is populated in the APM Infrastructure tab

### LINKED ISSUE #4359: [Request]: Document what products/integrations are able to populate the hosts/infra applications (e.g. also when embedded in APM)
### Description

It's not clear how some O11y tabs under APM can be populated.

E.g.
- Why my infrastructure tab under APM application is empty? 
- Why mi infrastructure tab under APM application is populated? E.g. https://demo.elastic.co/app/apm/services/productCatalogService/infrastructure?comparisonEnabled=true&environment=ENVIRONMENT_ALL&kuery=&latencyAggregationType=avg&offset=1d&rangeFrom=now-15m&rangeTo=now&serviceGroup=&transactionType=request
- What should the user do in order to populate those? Should they install/use:
    - Elastic APM Agents collecting metrics?
    - Metricbeat System module? Docker module? K8s module? Prometheus module?
    - EA with System integration? Docker integration? K8s integration? Prometheus integration?
    - OTel processors & OTel Collector? Which processors? On which version?
    - ...
- Are there additional information/metadata to add to the documents in order to benefit from correlation of APM & Metrics?

### Resources

We have some pointers at https://discuss.elastic.co/t/infrastructure-tab-in-the-apm-ui-app/323577/2

It would be nice to give hints to users on how to "populate" that parts or self-diagnose why they're empty.

Current doc is limited https://www.elastic.co/guide/en/observability/current/apm-infrastructure.html

Should we enrich and point to https://www.elastic.co/guide/en/observability/current/view-infrastructure-metrics.html ?

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

None

### What release is this request related to?

8.15

### Collaboration model

Other (please describe below)

### Point of contact.

**Main contact:** @roshan-elastic + @smith ?

**Stakeholders:**

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Description

Clarifies how data is populated in the Infrastructure tab for a selected service in the Applications UI. This PR also attempts to provide guidance on how to self-diagnose / troubleshoot when the Infrastructure tab does not have any data.

I searched through some GitHub issues and discussion threads in addition to the resources @lucabelluccini linked to in #4359, but I wasn't able to find much information on troubleshooting. 

This PR definitely needs more work, but I thought I'd open it to have something for the product/dev teams to respond to. @roshan-elastic can you help me find a technical reviewer? @lucabelluccini does this align with what you were expecting when you opened #4359?

### Documentation sets edited in this PR

_Check all that apply._

- [x] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue

Closes #4359

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Comment by roshan-elastic** [comment]
Hey @colleenmcginnis,

Thanks for this. Let me ask some of the team how this works - it was made a fair while ago.

@kpatticha / @jasonrhodes - Do you know who might be able to evaluate these docs which are explaining how the 'infrastructure' tab works in the service views?

[Preview - Infrastructure Tab](https://observability-docs_bk_4692.docs-preview.app.elstc.co/guide/en/observability/master/apm-infrastructure.html)
[Preview - No data shown in the infrastructure tab](https://observability-docs_bk_4692.docs-preview.app.elstc.co/guide/en/observability/master/apm-common-problems.html#troubleshooting-apm-infra-data)

**Comment by kpatticha** [comment]
@roshan-elastic 

@MiriamAparicio would you be able to take a look at these documents? I believe you’re the best person to provide input since you worked on the infrastructure tab in the service views.

 If your availability is limited, I’m happy to dig into the code and help out

**Comment by colleenmcginnis** [comment]
👋 @MiriamAparicio are you available to take a look at this PR? If not, @smith can you help me find a reviewer? Thanks!

**Comment by MiriamAparicio** [comment]
Hi @colleenmcginnis, sorry I missed this, I'll take a look tomorrow first thing

File: docs/en/observability/apm/view-and-analyze/drill-down/infrastructure.asciidoc [hunk 1]
```diff
@@ -4,11 +4,28 @@
 beta::[]
 
 The *Infrastructure* tab provides information about the containers, pods, and hosts
-that the selected service is linked to.
+that the selected service is linked to:
+
+// Does it make sense to add this here?
+* *Pods*: Uses the `kubernetes.pod.name` from the <<apm-data-model-metrics,APM metrics data streams>>.
```
Comment (MiriamAparicio) [inline docs/en/observability/apm/view-and-analyze/drill-down/infrastructure.asciidoc:10]: I think it's good information to know which fields are queried to get the infrastructure data in APM

**Review by MiriamAparicio — DISMISSED** [review]
LGTM, thanks 🌟

File: docs/en/observability/apm/troubleshooting/common-problems.asciidoc [hunk 2]
```diff
@@ -282,10 +283,42 @@ If the service map is not showing an expected connection between the client and
...
-
 This setting is necessary, for example, for cross-origin requests.
 If you have a basic web application that provides data via an API on `localhost:4000`,
 and serves HTML from `localhost:4001`, you'd need to set `distributedTracingOrigins: ['https://localhost:4000']`
 to ensure the origin is monitored as a part of distributed tracing.
 In other words, `distributedTracingOrigins` is consulted prior to the APM agent adding the
 distributed tracing `traceparent` header to each request.
+
+// This info is not confirmed, but hopefully provides a structure
+// for product/dev to react to and follow!
```
Comment (colleenmcginnis) [inline docs/en/observability/apm/troubleshooting/common-problems.asciidoc:294]: @MiriamAparicio Can you confirm that this is all correct too?

Comment (MiriamAparicio) [reply docs/en/observability/apm/troubleshooting/common-problems.asciidoc:294]: hi, yes this seems correct

Comment (colleenmcginnis) [reply docs/en/observability/apm/troubleshooting/common-problems.asciidoc:294]: Thanks!

**Review by bmorelli25 — APPROVED** [review]

**Review by roshan-elastic — APPROVED** [review]
LGTM!
